lib/model/CTokenListDataCategorizerBase.cc

```diff
@@ -467,6 +472,12 @@ void CTokenListDataCategorizerBase::addCategoryMatch(bool isDryRun,
     }
 }
 
+void CTokenListDataCategorizerBase::discardLatestTokens(std::size_t previousTokenCount) {
+    while (m_TokenIdLookup.size() > previousTokenCount) {
```

Comment:
I thought a random access `multi_index` has [resize](https://www.boost.org/doc/libs/1_53_0/libs/multi_index/doc/reference/rnd_indices.html)?

Reply:
It does, but that would require that `CTokenInfoItem` has a default constructor, or that we construct a `CTokenInfoItem` with some special values that would be copied if the container was resized to be bigger.  (Obviously it never would get copied, since we only resize to smaller sizes.)

Reply:
I had a look and `erase` would be a way to do this in one function call from this code.  But internally `erase` is implemented by moving the elements to be deleted to the end of the vector and then repeatedly calling `pop_back`, so it wouldn't be any faster than what's here now.